packages/vite-node/src/cli.ts

```diff
@@ -17,31 +17,40 @@ cli
   .option('-c, --config <path>', 'Use specified config file')
   .option('-w, --watch', 'Restart on file changes, similar to "nodemon"')
   .option('--options <options>', 'Use specified Vite server options')
+  .option('--script-mode', 'Use vite-node as a script runner')
   .help()
 
 cli
   .command('[...files]')
+  .allowUnknownOptions()
   .action(run)
 
 cli.parse()
 
 export interface CliOptions {
   root?: string
+  scriptMode?: string
   config?: string
   watch?: boolean
   options?: ViteNodeServerOptionsCLI
   '--'?: string[]
 }
 
 async function run(files: string[], options: CliOptions = {}) {
+  if (options.scriptMode)
+    files = [options.scriptMode]
```

Comment:
Script mode can only be used with one file? What if user calls it like this: `vite-node file --script-mode`?

Reply:
If we want to support that I'm afraid we have to drop `cac` and use something better at parsing args.
Right now `--script-mode` would only be used in shebang as you can already do `vite-node file --` if you want to call it from a shell.

Reply:
We can just add a validation here ourselves